pkgs/native_toolchain_c/test/clinker/objects_test.dart

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+//TODO(mosuem): Enable for windows and mac.
+@TestOn('linux')
+library;
+
+import 'dart:io';
+
+import 'package:native_assets_cli/native_assets_cli.dart';
+import 'package:native_toolchain_c/native_toolchain_c.dart';
+import 'package:native_toolchain_c/src/utils/run_process.dart';
+import 'package:test/test.dart';
+
+import '../helpers.dart';
+
+Future<void> main() async {
+  if (!Platform.isLinux) {
+    // Avoid needing status files on Dart SDK CI.
+    return;
+  }
+  const architecture = Architecture.x64;
+  const os = OS.linux;
+  const name = 'mylibname';
+
+  final ldConfig = CCompilerConfig(linker: Uri.file('/usr/bin/ld'));
```

Comment:
I mean testing if the link flags are correct for `ld` vs `clang` - we do support both on linux correct?

Reply:
`clang` is a compiler, `ld` is a linker. `clang` can automatically invoke `ld` and forwards the commandline options to `ld` https://clang.llvm.org/docs/ClangCommandLineReference.html#linker-options.

Are you saying the flags are different on both? And you want to test that we pass the right flags in both situations?
But our use cases don't support tree-shaking immediately on a `clang` invocation in a build hook currently. So we only need the `ld` flags to be tested right?
Or am I misunderstanding what you're trying to do here?

Reply:
> Are you saying the flags are different on both? And you want to test that we pass the right flags in both situations?

Yes and yes, the clang options are forward by prefixing `-Wl`, which is tested here.

> But our use cases don't support tree-shaking immediately on a `clang` invocation in a build hook currently.

But we do support using `clang` to treeshake in a link hook, which is tested here.

Reply:
> But we do support using `clang` to treeshake in a link hook, which is tested here.

Why don't we use `ld` instead of `clang` in the `CLinker`? What use case is `clang` supporting?

Reply:
I thought we want to give users the choice as to which linker they want to use. For example, they may instruct clang to use a linker other than `ld`.

Reply:
Not at the moment. flutter_tools decides what compiler, linker, and archiver to use. Not the users.

Reply:
This is for Flutter then, correct?

Anyhow, I will add a test with autodetecting, and run this on linux only :+1: